src/olympia/files/utils.py

```diff
@@ -198,14 +204,16 @@ def gecko(self):
         """Return the "applications|browser_specific_settings["gecko"]" part
         of the manifest."""
         parent_block = self.get(
-            'browser_specific_settings', self.get('applications', {})
+            'browser_specific_settings', self.get('applications', EMPTY_FALLBACK_DICT)
         )
-        return parent_block.get('gecko', {})
+        return parent_block.get('gecko', EMPTY_FALLBACK_DICT)
 
     @property
     def gecko_android(self):
         """Return `browser_specific_settings.gecko_android` if present."""
-        return self.get('browser_specific_settings', {}).get('gecko_android', {})
+        return self.get('browser_specific_settings', EMPTY_FALLBACK_DICT).get(
```

Comment:
The fallback of `gecko_android` should be `gecko` first.

Reply:
There is specific code for that when it comes to min/max version handling. We don't want to return `gecko` here because we need to distinguish from the case where `gecko_android` is present.